internal/daemon/worker/status.go

```diff
@@ -3,21 +3,24 @@ package worker
 import (
 	"context"
 	"errors"
+	"fmt"
 	"math/rand"
+	"sort"
+	"strings"
 	"time"
 
-	"github.com/hashicorp/boundary/internal/servers"
-
 	"github.com/hashicorp/boundary/internal/daemon/worker/common"
 	"github.com/hashicorp/boundary/internal/daemon/worker/session"
 	pbs "github.com/hashicorp/boundary/internal/gen/controller/servers/services"
 	"github.com/hashicorp/boundary/internal/observability/event"
+	"github.com/hashicorp/boundary/internal/servers"
 	"google.golang.org/grpc/resolver"
 )
 
 type LastStatusInformation struct {
 	*pbs.StatusResponse
-	StatusTime time.Time
+	StatusTime              time.Time
+	LastCalculatedUpstreams []resolver.Address
```

Comment:
Given that strings go over the RPC, it feels like the logic below could be simplified if this was `[]string` instead.